github/repos_pages.go

```diff
@@ -75,6 +75,53 @@ func (s *RepositoriesService) EnablePages(ctx context.Context, owner, repo strin
 	return enable, resp, nil
 }
 
+// PagesUpdate sets up parameters needed to update a GitHub Pages site.
+type PagesUpdate struct {
+	// CNAME represents a custom domain for the repository.
+	// Setting CNAME as an empty string will remove the custom domain.
+	CNAME *string `json:"cname,omitempty"`
+	// Source must include the branch name, and may optionally specify the subdirectory "/docs".
+	// Possible values are: "gh-pages", "master", and "master /docs".
+	Source *string `json:"source,omitempty"`
+}
+
+// pagesUpdateNullCNAME is used internally by UpdatePages to pass a null
+// value for the CNAME field when an empty string has been provided in
+// order to remove the custom domain.
+type pagesUpdateNullCNAME struct {
+	CNAME  *string `json:"cname"`
+	Source *string `json:"source,omitempty"`
+}
+
+// UpdatePages updates GitHub Pages for the named repo.
+//
+// GitHub API docs: https://developer.github.com/v3/repos/pages/#update-information-about-a-pages-site
+func (s *RepositoriesService) UpdatePages(ctx context.Context, owner, repo string, opts *PagesUpdate) (*Response, error) {
+	u := fmt.Sprintf("repos/%v/%v/pages", owner, repo)
+
+	req, err := s.client.NewRequest("PUT", u, opts)
+	if err != nil {
+		return nil, err
+	}
+
+	if *opts.CNAME == "" {
```

Comment:
Let's delete this block completely (lines 107-116) and remove the `pagesUpdateNullCNAME` and then simply remove the `,omitempty` on line 82.

That way, if `CNAME` is defined, it is used, but if `CNAME` is empty, `null` will be sent and the custom domain will be removed.

Reply:
Sure.